source/s3_auto_ranged_put.c

```diff
@@ -1017,8 +1112,10 @@ static void s_s3_prepare_upload_part_on_skipping_done(void *user_data) {
         return;
     }
 
-    /* Skipping succeeded */
-    auto_ranged_put->prepare_data.num_parts_read_from_stream = request->part_number - 1;
+    if (aws_future_get_bool(part_prep->skipping_future)) {
+        /* Skipping succeeded */
+        auto_ranged_put->prepare_data.num_parts_read_from_stream = request->part_number - 1;
+    }
```

Comment:
This new if-branch worries me

Before the `Content-Length` changes this, we always ran
`put->prepare_data.num_parts_read_from_stream = request->part_number - 1;`
after running s_skip_parts_from_stream() (even if it didn't actually skip anything)
and later after doing a successful read we incremented `num_parts_read_from_stream` again

The [Content-Length changes](https://github.com/awslabs/aws-c-s3/pull/285/files#diff-1b77484ee8493ab0ab8653a326f4bb0ac42a569e63261469b653ab544d27d99aR917) removed this line entirely, and simply did
`auto_ranged_put->prepare_data.num_parts_read_from_stream = request->part_number;`
after successfully reading from stream

so this weird way of sometimes updating the number here and sometimes not is weird.
let's go with the new way of just updating the number once, after we've read the part